api/representation.go

```diff
@@ -126,120 +120,108 @@ func (tr taskRequest) ToTaskRequestConfig(bp *config.BufferPeriodConfig, wd stri
 		}
 	}
 
-	tc.Finalize(bp, wd)
-	err := tc.Validate()
-	if err != nil {
-		return taskRequestConfig{}, err
-	}
-
-	var variables map[string]string
 	if tr.Variables != nil {
-		variables = tr.Variables.AdditionalProperties
-	}
-
-	trc := taskRequestConfig{
-		TaskConfig: tc,
-		variables:  variables,
+		tc.Variables = make(map[string]string)
+		for k, v := range tr.Variables.AdditionalProperties {
+			tc.Variables[k] = v
+		}
 	}
-	return trc, nil
+	return tc, nil
 }
 
 // String writes out the task request in an easily readable way
 // useful for logging
 func (tr taskRequest) String() string {
 	data, _ := json.Marshal(tr)
-	return fmt.Sprintf("%s", data)
+	return string(data)
 }
 
 type taskResponse oapigen.TaskResponse
 
-func taskResponseFromTaskRequestConfig(trc taskRequestConfig, requestID oapigen.RequestID) taskResponse {
+func taskResponseFromTaskConfig(tc config.TaskConfig, requestID oapigen.RequestID) taskResponse {
 	task := oapigen.Task{
-		Description: trc.Description,
-		Name:        *trc.Name,
-		Module:      *trc.Module,
-		Version:     trc.Version,
-		Enabled:     trc.Enabled,
-		WorkingDir:  trc.WorkingDir,
+		Description: tc.Description,
+		Name:        *tc.Name,
+		Module:      *tc.Module,
+		Version:     tc.Version,
+		Enabled:     tc.Enabled,
+		WorkingDir:  tc.WorkingDir,
 	}
 
-	if trc.variables != nil {
+	if tc.Variables != nil {
 		task.Variables = &oapigen.VariableMap{
-			AdditionalProperties: trc.variables,
+			AdditionalProperties: tc.Variables,
 		}
 	}
 
-	if trc.Providers != nil {
-		task.Providers = &trc.Providers
+	if tc.Providers != nil {
+		task.Providers = &tc.Providers
 	}
 
-	if trc.Services != nil {
-		task.Services = &trc.Services
+	if tc.Services != nil {
+		task.Services = &tc.Services
 	}
 
-	task.SourceInput = new(oapigen.SourceInput)
-	switch si := trc.SourceInput.(type) {
-	case *config.ServicesSourceInputConfig:
-		if len(si.Names) > 0 {
-			task.SourceInput.Services = &oapigen.ServicesSourceInput{
-				Names: &si.Names,
-			}
-		} else {
```

Comment:
Want to double check if we should handle `names` for `source_input "services"` (there's a similar handling for `condition "services"`)

Reply:
Nevermind! Just saw your comment. Looks like you're on it